File: hadoop/src/main/java/com/practice/bigdata/hadoop/spark/WordCount_HDFS.py
"""
 Launch Python Spark Shell : pyspark
 spark-sql
"""
myfile = sc.textFile("hdfs://ip-20-0-21-161.ec2.internal:8020/user/edureka_424232/spark_filesystem/wordcount/python/input_small/wordcountproblem_small");
counts = myfile.flatMap(lambda line: line.split(" ")).map(lambda word: (word,1)).reduceByKey(lambda v1,v2: v1 + v2);

counts.saveAsTextFile("hdfs://ip-20-0-21-161.ec2.internal:8020/user/edureka_424232/spark_filesystem/wordcount/python/output/wordcountproblem_small");

File: hadoop/src/main/java/com/practice/bigdata/hadoop/spark/WordCount_HDFS.scala

/**
* Launch Scala Spark Shell : spark-shell
*/

val hadoopConf = sc.hadoopConfiguration;
var hdfs = org.apache.hadoop.fs.FileSystem.get(hadoopConf);

val myfile = sc.textFile("hdfs://ip-20-0-21-161.ec2.internal:8020/user/edureka_424232/spark_filesystem/wordcount/scala/input_small/wordcountproblem_small");
myfile.collect;
val counts = myfile.flatMap(line => line.split(" ")).map(word => (word, 1)).reduceByKey(_+_);
counts.collect;

  
counts.saveAsTextFile("hdfs://ip-20-0-21-161.ec2.internal:8020/user/edureka_424232/spark_filesystem/wordcount/scala/output/wordcountproblem_small");
